components/ca-list.js

```diff
@@ -0,0 +1,722 @@
+     * follow the HATEOAS links to locate either the schema or data & render a HTML table matching the schema/data.
+     */
+    class List extends HTMLElement {
+        /**
+         * Executes when the element is first created
+         * @returns {undefined} nothing
+         */
+        createdCallback() {
+
+            // always render the search filter, but hide it via css unless data-searchable=true
+            this.filterEl = createElement(this, 'input', {
+                class: 'ca-list-filter',
+                type: 'text',
+                placeholder: 'Enter search terms to filter the list below'
+            });
+            this.filterEl.onkeyup = this.eventDelegate.bind(this);
+
+            // build the HTML table and keep a reference to the elements we need to update later
+            this.table = createElement(this, 'table', {});
+            this.colgroup = createElement(this.table, 'colgroup', {});
+            this.thead = createElement(this.table, 'thead', {});
+            this.tbody = createElement(this.table, 'tbody', {});
+
+            // capture click events and map them to internal UI event handler (sits at parent level and catches child clicks)
+            this.table.onclick = this.eventDelegate.bind(this);
+
+            // react to literal attributes
+            if (this.src !== '') this.loadSrc();
+            if (this.schema) this.createTable();
+            if (this.sortIndex > -1) this.sortTable();
+        }
+
+        // noinspection JSUnusedGlobalSymbols
+        /**
+         * Executes when the element is attached to the DOM
+         * @returns {undefined} nothing
+         */
+        attachedCallback() {
+            this.attached = true;
+        }
+
+        /**
+         * Executes when any attribute is changed on the element
+         * @param {string} attrName - the name of the attribute to have changed
+         * @param {string} oldVal - the old value of the attribute
+         * @param {string} newVal - the new value of the attribute
+         * @returns {undefined} nothing.
+         */
+        attributeChangedCallback(attrName, oldVal, newVal) {
+
+            // Only need to re-render the component if the attrbute value has changed
+            if (oldVal !== newVal) {
+
+                switch (attrName) {
+
+                    case 'src':
+                        this.loadSrc();
+                        break;
+
+                    case 'sort-index':
+                        this.sortTable();
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+        }
+
+        /** @return {string} url to schema or instance data containing HATEOAS links */
+        get src() {
+            return this.getAttribute('src') || '';
+        }
+
+        /** @param {string} value - url to schema or instance data containing HATEOAS links */
+        set src(value) {
```

Comment:
I know the old code doesn't have default params, but shouldn't we use this where non-breaking changes are obvious?
`set src(value = '')`